bin/fvwm-menu-desktop.1.in

```diff
@@ -134,152 +189,269 @@ fvwm-menu-desktop creates icons with the right size in $HOME/.fvwm/icons.
-                  --enable-mini-icons -size 16 --with-titles
+AddToMenu MenuFvwmRoot "XDG Menu" Popup XDGMenu
+PipeRead 'fvwm-menu-desktop'
 .EE
 .RE
 
-This generates a single KDE Preferences menu with top title
-"FvwmPref", mini icons with a size of 16x16 pixels and titles for each
-sub menu. But only if a menu with the name \fIpreferences\fR
-exists in the users or system wide Xdg directories. Else it's empty!
+The menu is created once when fvwm loads. Since menu creation can
+sometimes be slow, this could cause fvwm to take longer to load than
+one wants.
 
-To add the generated menu to the "Utilities" menu for example put the
-following to your fvwm config file (~/.fvwm/config or ~/.fvwm/.fvwm2rc):
+The menu is only generated when fvwm starts. If software
+is installed or removed you will have to select the 'Regenerate' option
+to rebuild the menu.
+
+One way to speed things up is to save the menu in a file and only generate
+the menu when 'Regenerate' is selected. To do this use \fB--regen-cmd\fR
+to call a custom function and write the menu to a file using a command like
 
 .RS
 .EX
-AddToMenu Utilities "Preferences Menu" Popup FvwmPref
-PipeRead 'fvwm-menu-desktop --desktop kde --menu-type preferences \\
-                            --title FvwmPref [other options]'
+fvwm-menu-desktop --regen-cmd XDGRegen > $FVWM_USERDIR/.XDGMenu
 .EE
 .RE
 
-If you think that \fBfvwm-menu-desktop\fR slows down your menu startup too
-much do not use \fBPipeRead\fR. Instead run \fBfvwm-menu-desktop\fR and
-redirect the menu to a file and read that file in your .fvwm2rc file.  
+Then add the following to the fvwm config file to define the function
+XDGRegen. The second to last line will generate the menu if the menu
+file doesn't exist when fvwm starts.
 
 .RS
 .EX
-AddToMenu MenuRoot "XDG Menus" Popup FvwmMenu
- ...
-
-AddToMenu FvwmMenu
-+ DynamicPopupAction Function FuncRecreateXdgMenus
- ...
-
-DestroyFunc FuncRecreateXdgMenus
-AddToFunc   FuncRecreateXdgMenus
-+ I Test (f  $[FVWM_USERDIR]/.menu) Read $[FVWM_USERDIR]/.menu
-+ I TestRC (!Match) PipeRead 'fvwm-menu-desktop > $[FVWM_USERDIR]/.menu \\
-                              &&  echo "Read $[FVWM_USERDIR]/.menu"'
+DestroyFunc XDGRegen
+AddToFunc XDGRegen
++ I PipeRead 'fvwm-menu-desktop --regen-cmd XDGRegen > \\
+    $[FVWM_USERDIR]/.XDGMenu; echo "Nop"'
++ I Read $[FVWM_USERDIR]/.XDGMenu
+
+Test (!f $[FVWM_USERDIR]/.XDGMenu) XDGRegen
+Read $[FVWM_USERDIR]/.XDGMenu
 .EE
 .RE
 
-or if you want to show the menus directly in a normal "Root" menu use
-this:
+Using \fBPipeRead\fR pauses fvwm to wait for the menu creation to complete.
+If this is an issue, you can run the menu generation in the background with
+Exec. The problem with this is you still have to wait for the generation to
+complete to Read the menu file and access the menu. You could use Schedule
+in the XDGRegen function and guess the amount of time required, or you
+could use the module \fBFvwmCommand\fR as follows
 
```

Comment:
This entire section is misleading.   Just use PipeRead -- it's not even worth mentioning.